types/state/DataAccount.go

```diff
@@ -0,0 +1,72 @@
+package state
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/AccumulateNetwork/accumulate/types"
+)
+
+type DataAccount struct {
```

Comment:
`DataAccount` is also defined in `protocol/types.yml`. That one should be removed if we're going to add it here.

Reply:
what is the preference, there is also state/types.yml.  There isn't anything special about the state that needs customization, so I can just remove this file and put it in state/types.yml if that is where things are headed.  The only specialization I suppose is the chain type id.

Reply:
I'm going to use the protocol version seems that is what the other ones are doing.

Reply:
I prefer generated types, because marshaling is boilerplate stuff, and with generated types we only have one thing to debug (the generator).